sdks/go/pkg/beam/io/pubsubio/pubsubio.go

```diff
@@ -52,32 +52,50 @@ func init() {
 
 // ReadOptions represents options for reading from PubSub.
 type ReadOptions struct {
+	Topic              string
 	Subscription       string
 	IDAttribute        string
 	TimestampAttribute string
 	WithAttributes     bool
 }
 
 // Read reads an unbounded number of PubSubMessages from the given
-// pubsub topic. It produces an unbounded PCollecton<*PubSubMessage>,
+// pubsub topic or subscription. It produces an unbounded PCollecton<*PubSubMessage>,
 // if WithAttributes is set, or an unbounded PCollection<[]byte>.
-func Read(s beam.Scope, project, topic string, opts *ReadOptions) beam.PCollection {
+func Read(s beam.Scope, project string, opts *ReadOptions) beam.PCollection {
```

Comment:
Since we're making a breaking change, lets go all in. Make it `opts ReadOptions` which avoids the value being nil (and saving us the check), while also requiring the value to be configured.